### DISCUSSION TITLE
how to programmatically add tags to posts?

### DISCUSSION THREAD

**Discussion by freder** [11ty/eleventy#1494] [Q&A]
hi,

`eleventy` noob here. I would like to programmatically add tags to posts. after spending a couple of hours and only half getting there (related: https://github.com/11ty/eleventy/issues/926), I'm not sure how to continue.
basically I want to extract all inline `#hashtags` from the markdown content and add them to the list of tags from the frontmatter.

I tried two things (all based on https://github.com/11ty/eleventy-base-blog):

1. created `./posts/posts.11tydata.js`:

```js
module.exports = {
	eleventyComputed: {
		tags: (data) => {
			return ['posts', 'asdf'];
		},
	}
};
```

the problems:
- `data` does not seem to contain the markdown content, so I'd have to read the markdown file myself _again_ to be able to extract any tags
- `http://localhost:8080/tags/` lists the injected tag, but `http://localhost:8080/tags/asdf` does _not_ work.

---

2. in `.eleventy.js`:

```js
// [...]
eleventyConfig.addCollection("tagList", function(collection) {
    let tagSet = new Set();
    collection.getAll().forEach(function(item) {
        if ("tags" in item.data) {
            // --------------------------------------------------
            // extract tags from content
            const inlineTags = extractInlineTags(item.template.frontMatter.content);
            item.data.tags = [
                ...item.data.tags,
                ...inlineTags,
            ];
            // --------------------------------------------------

            const tags = item.data.tags.filter(function(item) {
                switch(item) {
                    case "all":
                    case "nav":
                    case "post":
                    case "posts":
                    return false;
                }
                return true;
            });

            for (const tag of tags) {
                tagSet.add(tag);
            }
        }
    });

    return [...tagSet];
});
```

problems:
- I do have access to the markdown content here, but `http://localhost:8080/tags/asdf` still does not work

what's the correct way to achieve this?

thanks in advance!

**Comment by freder** [comment]
(3.) for a moment I was also considering to swap out the yaml parser entirely (or find a way to hook into the _"post-parse"_ state of the data processing pipeline), but from what I can tell there is no easy way to do this.

**Comment by freder** [comment]
also, similar to https://github.com/11ty/eleventy/discussions/1468

**Reply by binyamin** [reply]
Just letting you know, I've seen your question. I haven't had a chance to go through it thoroughly, though. I'm sorry you encountered this as a beginner. Good luck!
